ui/common/src/common.ts

```diff
@@ -15,21 +15,30 @@ export interface PropWithEffect<T> extends Prop<T> {}
 // like mithril prop but with type safety
 export const prop = <A>(initialValue: A): Prop<A> => {
   let value = initialValue;
-  const fun = function (v: A | undefined) {
+  return (v?: A) => {
     if (defined(v)) value = v;
     return value;
   };
-  return fun as Prop<A>;
 };
 
 export const propWithEffect = <A>(initialValue: A, effect: (value: A) => void): PropWithEffect<A> => {
   let value = initialValue;
-  const fun = function (v: A | undefined) {
+  return (v?: A) => {
     if (defined(v)) {
       value = v;
       effect(v);
     }
     return value;
   };
-  return fun as PropWithEffect<A>;
 };
+
+export const withEffect =
```

Comment:
Implemented `withEffect` for purposes of extending other `Prop` implementations, such as those found in `storage.ts`. I debated another way of doing this, which was to add an optional parameter to all of those storage prop functions for providing an effect. Happy to implement that way if it is preferred. This way felt cleaner on first pass.